ip-utils.c

```diff
@@ -12,13 +12,45 @@
  */
 
 #include <errno.h>
+#include <sys/socket.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
+#include <netdb.h>
 #include <net/if.h>
 #include <string.h>
+#include <unistd.h>
+
+#include <glib.h>
 
 #include "ip-utils.h"
 
+int janus_network_detect_local_ip(char *buf, size_t buflen) {
```

Comment:
Excellent feedback, thanks for looking this through! Yes, this method is currently a bit hammered in and needs to be aligned in style and scope. I'll follow your suggestion on how to fix that. Anyway, the purpose here is not to constrain anything, but just to get a valid IP address that can theoretically reach the outside, so that you can advertise it around (e.g., in the SDP), so I'd still keep a string as the return value somehow to keep things easier.